tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/LocalizationStringTest.cs

```diff
@@ -5,41 +5,46 @@
 using System.Globalization;
 using System.Threading;
 using System.Collections.Generic;
+using Xamarin.Localization.MSBuild;
 
 namespace Xamarin.iOS.Tasks {
 	[TestFixture]
 	public class LocalizationStringTest : TestBase {
-		[TestCase ("cs-CZ", "došlo k chybě: neznámý formát image")]
-		[TestCase ("de-DE", "Unbekanntes Imageformat.")]
-		[TestCase ("en-US", "Unknown image format.")]
-		[TestCase ("es-ES", "formato de imagen desconocido.")]
-		[TestCase ("fr-FR", "format d'image inconnu.")]
-		[TestCase ("it-IT", "Formato immagine sconosciuto.")]
-		[TestCase ("ja-JP", "の読み込みでエラーが発生しました: 画像の形式が不明です。")]
-		[TestCase ("ko-KR", "을(를) 로드하는 동안 오류 발생: 알 수 없는 이미지 형식입니다.")]
-		[TestCase ("pl-PL", "nieznany format obrazu.")]
-		[TestCase ("pt-BR", "formato de imagem desconhecido.")]
-		[TestCase ("ru-RU", "неизвестный формат изображения.")]
-		[TestCase ("tr-TR", "yüklenirken hata oluştu: Görüntü biçimi bilinmiyor.")]
-		[TestCase ("zh-CN", "时出错: 未知图像格式")]
-		[TestCase ("zh-TW", "時發生錯誤: 未知的映像格式。")]
-		public void AllSupportedTranslations (string culture, string errorMessage)
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("pl-PL")]
+		[TestCase ("pt-BR")]
+		[TestCase ("ru-RU")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void SpecificErrorTranslation (string culture)
 		{
+			// insert which error code you'd like to test
+			string errorCode = "E0007";
```

Comment:
Hi Sebastien, I did not explain my motivation very well. I am very open to feedback on how to make this better here and in future tests! When I updated the localization strings translations from the loc team previously, I made a little test to go with it that only tested one specific error code which was helpful in making sure that this error was being fired in different languages based on someone's locale. I thought this would be even more useful if we could select which error code is firing so that if I update a localization string, I can make sure it is being translated correctly and not firing in English in other locales. I did co-write a wiki page about how to add localization strings and had a part at the bottom about how to test and thought I would replace that with instructions to this test unless there is a better way to do this: https://github.com/xamarin/maccore/wiki/Localization

About the user having to manually enter the code: I had thought about trying to iterate through all the error codes and see if they are translated making this more of a automatic test, but the loc team can take up to a week to do the translations after someone adds new error codes. Perhaps this could be a good way to make sure people update their translated strings, but it would slow down someone from making a PR up to a week which I did not think was a good idea.. 

Please let me know if anyone has any recommendations?

Reply:
Part of your reply should be copied in the PR description (and in the commit message before merging).

>  Perhaps this could be a good way to make sure people update their translated strings,

Yes, that does sound like a good idea.

> but it would slow down someone from making a PR up to a week 

Maybe not. We can have a list of _pending_ translations and only report errors if they are not part of the list. That would not block PR, the authors would add the codes to the _pending_ list. We can also report (without error) pending members that were translated, so anyone updated the list would be able to keep it up to date (a bit like xtro force you to keep the data file clean).

That can be discussed (in a future weekly call) and implemented in a different PR.

> Also don't mind keeping the old test and adding this as another test.

I'd rather keep the existing test - it works so it's free 😄